File: Documents/Ocean Workspace/FlightManagmentSystem/flightSystem/src/app/services/flight.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Flight } from '../model/flight';
import { Observable } from 'rxjs';
@Injectable({
  providedIn: 'root'
})
export class FlightService {

  private baseUrl = 'http://localhost:8082/flight';

  constructor(private http: HttpClient) {}

  addFlight(flight: Flight): Observable<Flight> {
    return this.http.post<Flight>(`${this.baseUrl}/addFlight`, flight);
  }

  searchFlight(departCity: string, arrivalCity: string): Observable<Flight[]> {
    return this.http.get<Flight[]>(`${this.baseUrl}/searchByCity/${departCity}/${arrivalCity}`);
  }

  getByFlightNo(flightNo: number): Observable<Flight> {
    return this.http.get<Flight>(`${this.baseUrl}/findByNumber/${flightNo}`);
  }

  findAll(): Observable<Flight[]> {
    return this.http.get<Flight[]>(`${this.baseUrl}/findAll`);
  }

  findFlightsBetweenCitiesOnDate(departureCity: string, arrivalCity: string, departureDate: string): Observable<Flight[]> {
    return this.http.get<Flight[]>(`${this.baseUrl}/search/${departureCity}/${arrivalCity}/${departureDate}`);
  }

  deleteById(flightNo: number): Observable<boolean> {
    return this.http.delete<boolean>(`${this.baseUrl}/deleteById/${flightNo}`);
  }

  updateFlight(flight: Flight): Observable<Flight> {
    return this.http.put<Flight>(`${this.baseUrl}/update`, flight);
  }
}

File: Documents/Ocean Workspace/Flight Booking System/flightService/src/main/java/com/flight/service/FlightService.java
package com.flight.service;

import java.util.Date;
import java.util.List;

import com.flight.Exception.ArrivalDepartureSameException;
import com.flight.Exception.FlightAlreadyExistException;
import com.flight.Exception.FlightNotFoundException;
import com.flight.entity.Flight;

public interface FlightService {
	
	public Flight addFlight(Flight flight)throws ArrivalDepartureSameException, FlightAlreadyExistException;
	
	public List<Flight> searchFlightByCity(String departCity, String arrivalCity) throws FlightNotFoundException;
	
	public Flight findByNumber(long flightNo) throws FlightNotFoundException;
	
	public List<Flight> getAll();
	
	public boolean deleteById(long flightNo)throws FlightNotFoundException;
	
	public Flight updateFlight(Flight flight)throws FlightNotFoundException;
	
	public List<Flight> findFlightsBetweenCitiesOnDate(String departureCity, String arrivalCity, Date departureDate); 
        
    

	

}

